src/Chapter2.hs

```diff
@@ -727,7 +738,7 @@ value of the element itself
 🕯 HINT: Use combination of 'map' and 'replicate'
 -}
 smartReplicate :: [Int] -> [Int]
-smartReplicate l = error "smartReplicate: Not implemented!"
+smartReplicate l = concat (map (\x -> replicate x x) l)
```

Comment:
Great! 👍🏼 
If you want to write this function even more elegantly, you can use the standard function `concatMap` which is a combination of `concat` and `map` as the name suggests 🙂
After that, you could notice, that you would be able to eta-reduce on the last argument :)

Reply:
Oh wow, that's a really nice solution. I'll try it out.
Is there another way to do this other than using `concat`?

Reply:
You can implement it using `foldr`, and you will need to use `++` anyway. So these two solutions are based on the base approach 🙂

Reply:
Don